### PR TITLE
GRW-2288 - refact(ProductPage): unify layout

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Use a single layout instead of specific implementations for Mobile and Desktop
* General styling changes: Navigation links are now displayed after the page gets scrolled

It feels like a lot, but most of the changes are just  moving things around. Gonna do my best to spot the actual changes with comments in this PR.

## Justify why they are needed

Requested by design. The idea was to implement something like [this](https://cloud.protopie.io/p/07d4b9f748c935858ceb59e0?ui=true&scaleToFit=true&enableHotspotHints=true&cursorType=touch&mockup=true&bgColor=%23F5F5F5&playSpeed=1&playerAppPopup=true) where:

* Tabs gets discarted and only internal links are used
* Links navigation are displayed below VideoBlock

The last one is hard to accomplish on a scalable way. The compromise was to display navigation links based on window scroll percentage.

## Jira issue(s): [GRW-2288](https://hedvig.atlassian.net/browse/GRW-2288)

[GRW-2288]: https://hedvig.atlassian.net/browse/GRW-2288?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1833** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1833" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1833?utm_source=stack-comment).

File: apps/store/src/blocks/ProductPageBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  // TODO: Ideally padding that are currently spcing content from the edge of the page
+  // should be added by the container.
+  paddingInline: 0,
+
+  [mq.lg]: {
+    paddingInline: 0,
+  },
+})
+
+const OverviewGridArea = styled(GridLayout.Content)({
```
Comment (guilhermespopolin) [inline apps/store/src/blocks/ProductPageBlock.tsx:222]: This is new.

File: apps/store/src/blocks/ProductPageBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+const VariantSelector = () => {
+  return (
+    <VariantSelectorWrapper>
+      <StyledProductVariantSelector />
+    </VariantSelectorWrapper>
+  )
+}
+
+const FixedHeader = ({ children }: { children: ReactNode }) => {
```
Comment (guilhermespopolin) [inline apps/store/src/blocks/ProductPageBlock.tsx:107]: This is new

File: apps/store/src/blocks/ProductPageBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+      {blok.body.map((nestedBlock) => (
+        <StoryblokComponent blok={nestedBlock} key={nestedBlock._uid} />
+      ))}
+    </Main>
+  )
+}
+ProductPageBlock.blockName = 'product'
+
+const VariantSelector = () => {
```
Comment (guilhermespopolin) [inline apps/store/src/blocks/ProductPageBlock.tsx:99]: This is new

**Review by robinandeer — APPROVED** [review]
Nice, I think it looks great :D

[haseverything](https://app.graphite.dev/macros/HedvigInsurance)

Let us consider scrolling up to the top of the "coverage" section when you change the product variant.

**Review by gustaveen — APPROVED** [review]
Great work, feels much smoother now! 🤩

Just the alignment in desktops, but no rush to adjust it in this PR 

Screenshot 2023-03-07 at 15 10 31

**Pull request merged**